lib/Sema/CSDiagnostics.cpp

```diff
@@ -6229,8 +6229,10 @@ bool InvalidPackElement::diagnoseAsError() {
 }
 
 bool InvalidPackReference::diagnoseAsError() {
-  emitDiagnostic(diag::pack_reference_outside_expansion,
-                 packType);
+  auto patternType =
+      getPatternTypeOfSingleUnlabeledPackExpansionTuple(packType);
```

Comment:
One note because I'll forget and it could totally be done separately - both `InvalidPackReference` and `InvalidPackElement` constructors should use `resolveType(packType)` and `resolveType(packElementType)` accordingly instead of just `packType` and `packElementType`, otherwise we'd end up with a diagnostic that could have type variables in it and crash.